ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -20,26 +20,66 @@
  * A message info class that contains basic info about a message
  */
 public class MessageInfo {
+
+  public static final short ACCOUNTID_CONTAINERID_DEFAULT_VALUE = -1;
+
   private final StoreKey key;
   private final long size;
-  private final long expirationTimeInMs;
-  private final boolean isDeleted;
-  private final Long crc;
+  private long expirationTimeInMs = Utils.Infinite_Time;
+  private boolean isDeleted = false;
+  private Long crc = null;
+  private short accountId;
+  private short containerId;
+  private long operationTimeMs;
 
-  public MessageInfo(StoreKey key, long size, long expirationTimeInMs) {
-    this(key, size, false, expirationTimeInMs);
-  }
+  public static class MessageInfoBuilder {
```

Comment:
If you want the builder inside of `MessageInfo` we can just call the inner class `Builder`, so that we can call `new MessageInfo.Builder()` instead of `new MessageInfo.MessageInfoBuilder()`. You could also move this into its own file and change the constructor to package private like some of the other builders